packages/hurl/src/main.rs

```diff
@@ -305,17 +305,13 @@ fn create_cookies_file(
 "#
     .to_string();
 
-    // TODO: We only serialize the first file. I don't see any valid reason why we're not
-    // serializing every file see issue <https://github.com/Orange-OpenSource/hurl/issues/2537>
-    match runs.first() {
-        None => {
-            return Err(CliError::IO("Issue fetching results".to_string()));
-        }
-        Some(run) => {
-            for cookie in run.hurl_result.cookies.iter() {
-                s.push_str(&cookie.redact(secrets));
-                s.push('\n');
-            }
+    if runs.is_empty() {
+        return Err(CliError::IO("Issue fetching results".to_string()));
+    }
+    for run in runs.iter() {
+        for cookie in run.hurl_result.cookies.iter() {
+            s.push_str(&cookie.redact(secrets));
```

Comment:
As the Netscape file format for cookie support comment (`# some comment`) maybe we can add a preliminary  comment on the file used something like `# Cookies for file <tests_ok/test.hurl>`. This will help separating the different cookies.